core/federated/RTI/rti_remote.c

```diff
@@ -401,6 +400,18 @@ void handle_timed_message(federate_info_t* sending_federate, unsigned char* buff
                  fed->enclave.last_granted.time - start_time, fed->enclave.last_granted.microstep,
                  fed->enclave.last_provisionally_granted.time - start_time,
                  fed->enclave.last_provisionally_granted.microstep);
+    // If the message was larger than the buffer, we must empty out the remainder also.
+    size_t total_bytes_read = bytes_read;
+    while (total_bytes_read < total_bytes_to_read) {
+      bytes_to_read = total_bytes_to_read - total_bytes_read;
+      if (bytes_to_read > FED_COM_BUFFER_SIZE) {
+        bytes_to_read = FED_COM_BUFFER_SIZE;
+      }
+      read_from_socket_fail_on_error(&sending_federate->socket, bytes_to_read, buffer, NULL,
```

Comment:
Should this really be fail on error?  It seems that if the fed is no longer connected, this is very likely to fail, and then the RTI will crash.

Reply:
Good point!

Reply:
We are not receiving data _from_ the disconnected fed. But from another normal fed which wants to send data to the disconnected one. 

But, should the RTI exit with error on socket errors? Probably not, but we have `read_from_socket_fail_on_error` all over the place. I am not sure exactly what it should do if it is not able to read out this entire message, because if TCP is working then it should come eventually, or the other fed or even the network failed between two chunks. 

I think this will require quite careful thought and a pass through the entire RTI and federated code-base to make sure that we can handle different scenarios. So I think for this small change we should keep to current style of the RTI (which is exiting if reading/writing from sockets fail), and then do a more thorough well-thought change later?

Reply:
Ah, yes, you are right.